packages/base-map/src/index.story.tsx

```diff
@@ -0,0 +1,175 @@
+
+import BaseMap, { MarkerWithPopup, LayerWrapper } from ".";
+
+import { withMap } from "../../../.storybook/base-map-wrapper";
+
+import "maplibre-gl/dist/maplibre-gl.css";
+
+const center: [number, number] = [45.522862, -122.667837];
+
+export default {
+  args: { center },
+  component: BaseMap,
+  decorators: [withMap()],
+  title: "BaseMap"
+};
+
+// eslint-disable-next-line react/jsx-props-no-spreading
+const Template = args => <BaseMap {...args} />;
+
+const samplePopup = (
+  <div>
+    <h1>Popup Title</h1>
+    <p>
+      Sample <span style={{ color: "purple" }}>popup</span> content.
+    </p>
+  </div>
+);
+
+const sampleMarkers = (
+  <MarkerWithPopup
+    tooltipContents={samplePopup}
+    position={center}
+  ></MarkerWithPopup>
+);
+
+const onClick = action("onClick");
+const onContextMenu = action("onContextMenu");
+const onViewportChanged = action("onViewportChanged");
+const a11yOverrideParameters = {
+  a11y: { config: { rules: [{ id: "color-contrast", reviewOnFail: true }] } }
+};
+
+export const clickAndViewportchangedEvents = Template.bind({});
+clickAndViewportchangedEvents.args = {
+  center,
+  onClick,
+  onContextMenu,
+  onViewportChanged
+};
+
+export const zoomed = Template.bind({});
+zoomed.args = {
+  center,
+  zoom: 17
+};
+
+const SetBoundsButton = () => {
+  const { mapWithBounds } = useMap();
+  const bbox = [-79, 43, -73, 45];
+
+  return (
+    <button
+      onClick={
+        () =>
+          mapWithBounds?.fitBounds(bbox, {
+            duration: 300,
+            padding: {
+              bottom: 25,
+              left: 15,
+              right: 5,
+              top: 10
+            }
+          })
+        // eslint-disable-next-line react/jsx-curly-newline
+      }
+    >
+      Set the bounds
+    </button>
+  );
+};
+
+export const clickToSetBounds = (): ComponentStory<typeof BaseMap> => (
+  <MapProvider>
+    <SetBoundsButton />
+    <BaseMap center={center} id="mapWithBounds" />
+  </MapProvider>
+);
+
+export const maxZoom = Template.bind({});
+maxZoom.args = {
+  maxZoom: 18,
+  zoom: 30
+};
+
+export const withCustomBaseLayer = Template.bind({});
+
+withCustomBaseLayer.args = {
+  baseLayer: `https://api.maptiler.com/maps/voyager/style.json?key=<PASTE YOUR MAPTILER TOKEN HERE>`,
+  center
+};
+
+export const withSampleMarkers = () => (
+  <BaseMap center={center}>{sampleMarkers}</BaseMap>
+);
+
+export const overlayWithLargeDataSet = () => (
+  <>
+    <div style={{ position: "relative", zIndex: 1000 }}>
+      Do not add Storybook overhead on layers with large dataset...
+    </div>
+    <BaseMap center={center}>
+      <AllVehiclesOverlay />
+    </BaseMap>
+  </>
+);
+
+export const customLocationPopupContent = () => (
+  <BaseMap center={center}>
+    <Popup longitude={center[1]} latitude={center[0]}>
+      {samplePopup}
+    </Popup>
+  </BaseMap>
+);
+export const optionalLayers = () => (
+  <BaseMap center={center}>
+    <LayerWrapper
```

Comment:
Just to check, is it intended that you can activate the hidden layer but not hide it again?

Reply:
No! This is a regression from when we moved the hiddenLayer updating to `useCallback`. I've reverted this change. The problem was the state that the `useCallback` method depended on was changing from inside the callback, so the callback never fired the update.

Reply:
Thanks for catching that